sql/parser/eval.go

```diff
@@ -65,6 +68,21 @@ func (d DBool) Type() string {
 	return "bool"
 }
 
+// Compare implements the Datum interface.
+func (d DBool) Compare(other Datum) int {
+	v, ok := other.(DBool)
```

Comment:
yeah, and that the same is true for all `Datum`s, meaning sort order is only partially defined